src/SerializerFactory.php

```diff
@@ -69,7 +69,8 @@ private function shouldUseObjectsForMaps() {
 	 * @return Serializer
 	 */
 	public function newEntitySerializer() {
-		$fingerprintSerializer = new FingerprintSerializer( $this->shouldUseObjectsForMaps() );
+		$options = $this->shouldUseObjectsForMaps() ? FingerprintSerializer::USE_OBJECTS_FOR_MAPS : FingerprintSerializer::USE_ARRAYS_FOR_MAPS;
```

Comment:
That'd make the serializers depend on a factory, which is not nice, even when ignoring the cyclic dependency. We could create a dedicated enum. But perhaps that ends up overdoing it?

Again, Python does not have this issue at all... :|